features/steps/steps.py

```diff
@@ -202,6 +229,28 @@ def step_impl(context, constraint, num, entity):
     handle_errors(context, errors)
 
 
+@then('Each instance of {entity} has a unique value for the attribute {attribute}')
+def step_impl(context, entity, attribute):
+    ent_attr = {'IfcShapeRepresentation':'Representations'}
+    assert entity in ent_attr
+    attr_get = ent_attr[entity]
+
+    errors = []
+    if context.instances:
```

Comment:
The if statement is not necessary. You can iterate over an empty list. As long as `context.instances` is not None which we can make sure it's not.